pandas/plotting/_core.py

```diff
@@ -889,6 +909,68 @@ def _make_plot(self):
             ax.errorbar(data[x].values, data[y].values,
                         linestyle='none', **err_kwds)
 
+    def _sci_notation(self, num):
+        """
+        Returns mantissa and exponent of the number passed in argument.
+        Example:
+        >>> _sci_notation(89278.8924)
+        (8.9, 5.0)
+        """
+        scientific_notation = '{:e}'.format(num)
+        expnt = float(re.search(r'e([+-]\d*)$',
+                                scientific_notation).groups()[0])
+        coef = float(re.search(r'^([+-]?\d\.\d)',
+                               scientific_notation).groups()[0])
+        return coef, expnt
+
+    def _legend_bubbles(self, s_data_max, size_factor, bubble_points):
+        """
+        Computes and returns appropriate bubble sizes and labels for the
+        legend of a  bubble plot. Creates 4 bubbles with round values for the
+        labels, the largest of which is close to the maximum of the data.
+        """
+        coef, expnt = self._sci_notation(s_data_max)
+        labels_catalog = {
```

Comment:
Where were these values taken from? Some comments may be helpful

Reply:
The values were defined by testing until I got something visually pleasing, similar to this graph with R in the original issue submitted here:
https://github.com/pandas-dev/pandas/issues/16827